filters/auth/oidc_test.go

```diff
@@ -137,3 +139,103 @@ func TestExtractDomainFromHost(t *testing.T) {
 		})
 	}
 }
+
+func TestNewOidc(t *testing.T) {
+	for _, tt := range []struct {
+		name string
+		args string
+		f    func(string) filters.Spec
+		want *tokenOidcSpec
+	}{
+		{
+			name: "test UserInfo",
+			args: "/foo",
+			f:    NewOAuthOidcUserInfos,
+			want: &tokenOidcSpec{typ: checkOIDCUserInfo, SecretsFile: "/foo"},
+		},
+		{
+			name: "test AnyClaims",
+			args: "/foo",
+			f:    NewOAuthOidcAnyClaims,
+			want: &tokenOidcSpec{typ: checkOIDCAnyClaims, SecretsFile: "/foo"},
+		},
+		{
+			name: "test AllClaims",
+			args: "/foo",
+			f:    NewOAuthOidcAllClaims,
+			want: &tokenOidcSpec{typ: checkOIDCAllClaims, SecretsFile: "/foo"},
+		},
+	} {
+		t.Run(tt.name, func(t *testing.T) {
+			if got := tt.f(tt.args); !reflect.DeepEqual(got, tt.want) {
+				t.Errorf("Failed to create object: Want %v, got %v", tt.want, got)
+			}
+		})
+	}
+
+}
+
+func TestCreateFilter(t *testing.T) {
+	for _, tt := range []struct {
+		name    string
+		args    []interface{}
+		want    filters.Filter
+		wantErr bool
+	}{
+		{
+			name:    "test no args",
+			args:    nil,
+			wantErr: true,
+		},
+		{
+			name:    "test wrong number of args",
+			args:    []interface{}{"s"},
+			wantErr: true,
+		},
+		{
+			name:    "test wrong number of args",
+			args:    []interface{}{"s", "d"},
+			wantErr: true,
+		},
+		{
+			name:    "test wrong number of args",
+			args:    []interface{}{"s", "d", "a"},
+			wantErr: true,
+		},
+		{
+			name:    "test wrong args",
+			args:    []interface{}{"s", "d", "a", "f"},
+			wantErr: true,
+		},
+		// cert error for provider
```

Comment:
is this a TODO comment?

Please use example domains, like example.org, because identity-provider.com and callback.com are existing, unrelated domain names.

Reply:
I would like to have a test case that also tests the happy path but I was not able to think about how to achieve it in this case. That's why it's commented.
Drop it?

Reply:
if I understand correctly, this would require a quite elaborate mocking. Let's drop it.

Reply:
done